cotel/src/main/scala/day1/InverseCaptcha.scala

```diff
@@ -0,0 +1,21 @@
+package day1
+
+object InverseCaptcha {
+
+  def captcha(captcha: String): Int = {
+    val parsedCaptcha = captcha.toList.map(_.asDigit)
+
+    if (!hasDuplicatedElements(parsedCaptcha)) 0
+    else recursiveCaptcha(parsedCaptcha :+ parsedCaptcha.head)
+  }
+
+  private def hasDuplicatedElements(captcha: List[Int]): Boolean =
+    captcha.distinct.size != captcha.size
```

Comment:
You could check first if size is <= 10, if it is not, for sure there are duplicates, can be handy for a better eficiency on very long lists.

Reply:
You are right with that property but we need to check if there are repeated numbers in a captcha of size `< 10`. This is a good solution for both cases and I don't think it would really have an impact on performance as the items being compared are primitive types and this problem is not really though for very long lists.

We could make a benchmark for comparing our solutions but I don't think I have that much type 😥

Reply:
Yeah, if size is <10 you check it, but if its not you don't have to do any iteration over the list  ❤️  O(k) ❤️ . But anyway, you are right, it's just a marginal improvement 😄